### PR TITLE
ARROW-7289: [C#] ListType constructor argument is redundant

### PR DESCRIPTION
**Pull request by HashidaTKS**
ListType constructor is redundant and should be seperated.

### CODE REVIEW

**Comment by HashidaTKS** [comment]
sorry, I'll fix the commit history.

**Comment by HashidaTKS** [comment]
I fixed it.

File: csharp/src/Apache.Arrow/Types/ListType.cs [hunk 1]
```diff
@@ -25,10 +25,16 @@ public sealed class ListType : ArrowType
...
         public IArrowType ValueDataType { get; }
 
-        public ListType(Field valueField, IArrowType valueDataType)
+        public ListType(Field valueField)
         {
             ValueField = valueField ?? throw new ArgumentNullException(nameof(valueField));
+            ValueDataType = ValueField.DataType;
+        }
+
+        public ListType(IArrowType valueDataType)
```
Comment (eerhardt) [inline csharp/src/Apache.Arrow/Types/ListType.cs]: Can this constructor just call into the constructor above? Similar to the way it works in C++:

https://github.com/apache/arrow/blob/81909dc0775a686cb8d89037dc96d2cd78e01a1e/cpp/src/arrow/type.h#L591-L597

File: csharp/src/Apache.Arrow/Types/ListType.cs [hunk 2]
```diff
@@ -25,10 +25,16 @@ public sealed class ListType : ArrowType
...
+        {
             ValueDataType = valueDataType ?? NullType.Default;
```
Comment (kou) [inline csharp/src/Apache.Arrow/Types/ListType.cs]: Should we throw an exception for `null` `valueDataType`?
`NullType` isn't usable.

@eerhardt What do you think about this?

File: csharp/test/Apache.Arrow.Tests/TypeTests.cs [hunk 3]
```diff
@@ -96,6 +96,23 @@ public void TestStructGetFieldByName()
...
+        [Fact]
+        public void TestListTypeConstructor()
+        {
+            var stringField = new Field.Builder().Name("item").DataType(StringType.Default).Build();
+            var stringType1 = new ListType(stringField);
+            var stringType2 = new ListType(StringType.Default);
+
+            var int64Field = new Field.Builder().Name("item").DataType(Int64Type.Default).Build();
+            var int64Type1 = new ListType(int64Field);
+            var int64Type2 = new ListType(Int64Type.Default);
```
Comment (kou) [inline csharp/test/Apache.Arrow.Tests/TypeTests.cs]: It seems that `int64` case is redundant.
Should we check both `string` and `int64` cases?

**Review by kou — COMMENTED** [review]
It almost looks good to me.
Could you also check my comments?

File: csharp/src/Apache.Arrow/Types/ListType.cs (hunk 1 above)
Comment (HashidaTKS) [reply csharp/src/Apache.Arrow/Types/ListType.cs]: Thanks for your suggestion.
I modified it.

File: csharp/test/Apache.Arrow.Tests/TypeTests.cs (hunk 3 above)
Comment (HashidaTKS) [reply csharp/test/Apache.Arrow.Tests/TypeTests.cs]: Indeed, I remove it.

File: csharp/src/Apache.Arrow/Types/ListType.cs (hunk 2 above)
Comment (HashidaTKS) [reply csharp/src/Apache.Arrow/Types/ListType.cs]: The line was removed by the new commit but same check exists in Field.
Should we throw an exception there?

https://github.com/apache/arrow/blob/master/csharp/src/Apache.Arrow/Field.cs#L44

Comment (eerhardt) [reply csharp/src/Apache.Arrow/Types/ListType.cs]: I don't see this behavior in either the C++ or Java implementation.

I'd say it is up to you whether to leave the Field constructor as-is, or change it to throw on a `null` type. I could go either way on it.

**Review by eerhardt — APPROVED** [review]
Looks good to me. Thanks @HashidaTKS!

File: csharp/src/Apache.Arrow/Types/ListType.cs (hunk 2 above)
Comment (kou) [reply csharp/src/Apache.Arrow/Types/ListType.cs]: OK. We can do it as a follow-up task when we want to do.
We can merge this.

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**